src/scheduler.py

```diff
@@ -197,10 +197,10 @@ def UpdateClusterState(self, request, context):
                 updated_state['mps'].num_available -= 1
         # Log completed runs to retrieve stats post processing from kafka
         elif request.reason == 'end mix':
-            echo_cmd = f"echo {request.run_id} >> /tmp/tiebreaker-completed-mixes.txt"
+            echo_cmd = f'[ -p "/tmp/{request.run_id}.client-wait" ] && echo "1" > /tmp/{request.run_id}.client-wait'
```

Comment:
Why this change?

Reply:
This is the mkfifo change I had proposed. But in earlier proposal there was a blocking nature as I suggested the scheduler to create the fifo file. Now in this change, the client creates the fifo file. If the file exists the scheduler echoes there by there is no blocking nature.

Reply:
Ah okay, makes sense.